### PR TITLE
Add ignore_index in cross_entropy op

### PR DESCRIPTION
**Pull request by baiyfbupt**

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -1012,7 +1012,9 @@ def cross_entropy(input, label, soft_label=False):
                                tensor<float/double> with shape [N x D].
         soft_label (bool): a flag indicating whether to
                                            interpretate the given labels as soft
-                                           labels, default `False`.
+                                           labels. Default: `False`.
+        ignore_index (int): Specifies a target value that is ignored and does not
+                            contribute to the input gradient. Default: -100
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: Need to tell uers this argument is only worked when soft_label=False.

**Review by qingqing01 — COMMENTED** [review]
also need to enhance softmax_with_cross_entropy_op

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (baiyfbupt) [reply python/paddle/fluid/layers/nn.py]: Done

**Comment by baiyfbupt** [comment]
@Done

**Comment by baiyfbupt** [comment]
@qingqing01 Done

**Review by qingqing01 — APPROVED** [review]

**Comment by qingqing01** [comment]
@panyx0718 This PR changes the API of cross_entropy and softmax_cross_entropy,  add `ignore_index`,  similar to PyTorch https://pytorch.org/docs/stable/nn.html?highlight=cross_en#crossentropyloss.   Please help to reveiw, Thanks!

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -1012,7 +1012,10 @@ def cross_entropy(input, label, soft_label=False):
                                tensor<float/double> with shape [N x D].
         soft_label (bool): a flag indicating whether to
                                            interpretate the given labels as soft
-                                           labels, default `False`.
+                                           labels. Default: `False`.
+        ignore_index (int): Specifies a target value that is ignored and does
```
Comment (panyx0718) [inline python/paddle/fluid/layers/nn.py:1016]: Is there other way to mask label with some op? so gradient doesn't get propagated? I don't like this way of solution. What if you need to ignore multiple index in the future?

**Review by panyx0718 — APPROVED** [review]
